### PR TITLE
Generate SSH key pair on demand, download it using JavaScript

### PR DESCRIPTION
**Pull request by Kami**
This pull request updates KeyPair controller to create SSH key pair on demand instead of generating it inside the constructor and leaving it hanging in the memory...

In addition to that, it implements "download key as file" functionality on the client side so we don't need to leave key hanging in memory. If we want to use old approach, but do it more securely we would need to implement some kind of session mechanism, but this is a big over kill since we can just handle it with JavaScript (all modern browsers support this functionality).

### CODE REVIEW

File: public/scripts/main.js [hunk 1]
```diff
@@ -504,4 +504,33 @@ $(function() {
   if ($('#page-puppet').length) {
     puppet.init();
   }
+
+  $(document).on('click', 'a#download-public-key', function() {
```
Comment (Kami) [inline public/scripts/main.js]: Yeah, it's not the nicest code, but it works :P

File: public/scripts/main.js [hunk 2]
```diff
@@ -504,4 +504,33 @@ $(function() {
...
+  });
+
+  $(document).on('click', 'a#download-private-key', function() {
+    var filename, data;
+    filename = 'st2-ssh.key';
+    data = $('#keypair-private').val();
+    elem = download_as_file(filename, data);
+    $('body').append(elem);
+    elem.click();
+  });
```
Comment (emedvedev) [inline public/scripts/main.js]: I'd really make this one function instead of reusing what essentially is the same code twice.

Comment (emedvedev) [inline public/scripts/main.js]: +1 on everything else.

**Comment by Kami** [comment]
Since we now generate the key on demand, clicking on "Next" in the second step will result in a short delay while key is being generated so we should display some kind of loading animation.

I would prefer someone else which is more up to data with JavaScript to do that :)

**Comment by dennybaa** [comment]
If I understood right, it's in that dialog with pubkey and prikey. Would be also nice to have copy to clipboard button!

File: public/scripts/main.js (hunk 2 above)
Comment (Kami) [inline public/scripts/main.js]: d68825ca59ab6c38e896b0e0455db9e527f7ed2d

Gosh, haven't done JS for a while, it's still terrible :D

**Comment by Kami** [comment]
@dennybaa Yeah, but that's really a new feature so I will leave this to someone else when time permits (my JS is really rusty).

**Comment by Kami** [comment]
Updated affected test case, added a new one. Merging.

**Comment by Kami** [comment]
@emedvedev When you have time can you please have a look at adding spinner / loading animation when clicking next on step 2?

**Comment by emedvedev** [comment]
Yes sir.

**Comment by armab** [comment]
@Kami Elegant solution :+1: 

Is it available for manual testing, or needs merging into `stable` branch first?

**Comment by Kami** [comment]
@armab Yeah, you can clone the repo and follow the instructions in README (create virtualenv, install requirements in it, use uwsgi to run the app), that's also how I tested it.

**Pull request merged**